interop-monix/shared/src/test/scala/zio/interop/monix/MonixTaskSpec.scala

```diff
@@ -0,0 +1,162 @@
+          } yield ((orig, current))
+          assertM(io)(equalTo((0, 1)))
+        },
+        testM("the ZIO task fails if a Monix async task fails") {
+          val error     = new Exception("async monix failure")
+          val monixTask = MTask.async[Int] { cb =>
+            cb.onError(error)
+          }
+          val io        = ZIO.fromMonixTask(monixTask)
+          assertM(io.either)(isLeft(equalTo(error)))
+        },
+        testM("lazily executes a converted Monix task") {
+          @volatile var executed = false
+          val monixTask          = MTask.eval {
+            executed = true
+          }
+          val test               = ZIO.succeed {
+            executed = false
+          } *> {
+            ZIO.fromMonixTask(monixTask)
+            ZIO.succeed(executed)
+          }
+          assertM(test)(isFalse)
+        },
+        testM("propagates cancellation from ZIO to Monix") {
+          @volatile var cancelled = false
+          @volatile var running   = false
+          val monixTask           = MTask.eval {
+            running = true
+          } >> MTask.never.doOnCancel {
+            MTask.eval {
+              cancelled = true
+            }
+          }
+          for {
+            _            <- ZIO.succeed {
+                              cancelled = false
+                              running = false
+                            }
+            fiber        <- ZIO.fromMonixTask(monixTask).fork
+            // wait until the monix task is running before interrupting the fiber
+            _            <- ZIO.succeed(running).repeatUntil(Predef.identity)
+            exit         <- fiber.interrupt
+            wasCancelled <- ZIO.succeed(cancelled)
+          } yield assertTrue(wasCancelled) && assertTrue(exit.interrupted)
+        } @@ nonFlaky
+      ),
+      suite("ZIO#toMonixTask")(
+        testM("converts a successful ZIO task to a Monix task") {
+          @volatile var testVar = 0
+          val io                = ZIO.succeed {
+            val old = testVar
+            testVar = old + 1
+            old
+          }
+          for {
+            monixTask <- io.toMonixTask
+            result  <- ZIO.fromMonixTask(monixTask)
+            current <- ZIO.succeed(testVar)
+          } yield assertTrue(result == 0) && assertTrue(current == result + 1)
+        },
+        testM("converts a failed ZIO task to a failed Monix task") {
+          val error = new Exception("ZIO operation failed")
+          val io    = ZIO.fail(error)
+          val test  = io.toMonixTask.flatMap[Any, Throwable, Nothing](ZIO.fromMonixTask(_)).either
+          assertM(test)(isLeft(equalTo(error)))
+        },
+        testM("propagates cancellation from Monix to ZIO") {
+          @volatile var cancelled = false
+          @volatile var running   = false
+          val io                  = ZIO.succeed {
+            running = true
+          } *> ZIO.never.onInterrupt {
+            ZIO.succeed {
```

Comment:
```suggestion
            ZIO.effectTotal {
```

Reply:
I somehow got the idea that in ZIO 2, `succeed` was preferred.